### DISCUSSION TITLE
Seeking advice on generic `notes` column vs table

### DISCUSSION THREAD

**Discussion by nexushoratio** [sqlalchemy/sqlalchemy#12069] [Usage Questions] [answered]
This is more of a general database design issue, but 1) lots of smart folks here and 2) sqlalchemy might have some helpful features I haven't discovered yet.

I have a toy commandline app with nine tables, with a few hundred thousand rows.  Currently it uses sqlite, but I'd like to keep it generic.  After all, this toy is as much about building skills as it is about using it.

I'm finding myself wanting to add a generic text field to some tables that the user (well, so far that's just me), can add notes about the entry.  Of course, whenever I decide to do this after the fact, it requires a schema change, which is a tiny bit annoying.

I'm wondering if it makes more sense to have a notes table instead, and if so, what would be some good ways of tying them back to the original row.

Since it would cover multiple tables, FKs are out, so I need either triggers on each table, or cleaning routine that gets run regularly.  And each table I might want to support has a different type of PK anyway.

If I was using a more classic database that had fixed lengths for text columns, this would be less of a decision.  I would essentially use the a notes table as a text blob with columns of (id, segment, content), then the app would concatenate the content in segment order on read, and split upon write.  But I think most DBs don't have to worry about that these days?  (Or maybe even that is too old school?)

Anyway, if anyone has any advice, or examples of what they've done they'd be willing to share, that would be cool.

Thanks!

**Comment by CaselIT** [comment, marked as answer]
Hi,

Personally I would go with the column in the tables that needs it. Adding a complete table seems a bit overkill for that use case.
The table may make sense if it becomes an entity in your application, so that an user can add a note and then connect that note to different other tables.

But that's just my 2 cents here.
